torch/csrc/jit/tensorexpr/operators/unary.cpp

```diff
@@ -0,0 +1,39 @@
+#include <torch/csrc/jit/tensorexpr/operators/unary.h>
+
+namespace torch {
+namespace jit {
+namespace tensorexpr {
+
+using namespace torch::jit::tensorexpr;
+
+Tensor computeSign(
+    const std::vector<ArgValue>& inputValues,
+    const std::vector<ExprHandle>& outputShape) {
+  return Compute(
+      "aten_sign", c10::fmap<DimArg>(outputShape), [&](ParameterList& axes) {
+        std::vector<ExprHandle> indices(axes.begin(), axes.end());
+        std::vector<ExprHandle> inputs = {
+            tensorOrConstant(inputValues[0], indices)};
+        auto inp = promoteIntegerToDefaultType(inputs[0]);
+        // v1 = (inp < 0.0f) ? -1.0f : 1.0f;
+        auto v1 = CompareSelect::make(
+            inp, ExprHandle(0.0f), ExprHandle(-1.0f), ExprHandle(1.0f), kLT);
+        // v2 = (inp == 0.0f) ? 0.0f : v1
+        auto v2 = CompareSelect::make(
+            inp, ExprHandle(0.0f), ExprHandle(0.0f), v1, kEQ);
+        // (isnan(inp) == 1) ? 0.0f : v2
+        auto res = CompareSelect::make(
+            isnan(inp), ExprHandle(1), ExprHandle(0.0f), v2, kEQ);
+        // Final expression:
+        //    (isnan(inp) == 1) ?
+        //        0.0f :
+        //        (inp == 0.0f) ?
+        //            0.0f :
+        //            (inp < 0.0f) ? -1.0f : 1.0f
+        return res;
+      });
```

Comment:
A couple issues here:
1. I'd probably use the same expression as ATen `(0<a) - (a<0)`, https://github.com/pytorch/pytorch/blob/a8ea3461145551bb85855135eafb84fc754fd320/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L299
2. I'm not sure why this doesn't fail in unit tests, but the dtypes in the `CompareSelects` aren't going to be right since you're explicitly creating floats and comparing to whatever the input dtype is (so, double would probably break).  Instead of `ExprHandle(-1.0f)` you should probably do `immLike(inp, -1.0f)`.

Reply:
1. Changed the implementation to the one in ATen (I had initially thought of doing that as a follow up..)

2. The unit test wasn't failing because it only had `float` inputs. I updated the test to add `double` as well, and fixed the implementation accordingly.